x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_properties/index_check_fields/index.tsx

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo, useState } from 'react';
+import { EuiButtonGroup, EuiFlexGroup, EuiSpacer } from '@elastic/eui';
+import styled from 'styled-components';
+
+import { useDataQualityContext } from '../../data_quality_context';
+import { useIndicesCheckContext } from '../../../contexts/indices_check_context';
+import { EMPTY_METADATA, INCOMPATIBLE_TAB_ID } from '../helpers';
+import { IlmPhase, PatternRollup } from '../../../types';
+import { getTabs } from '../../tabs/helpers';
+
+const StyledTabFlexGroup = styled(EuiFlexGroup)`
+  width: 100%;
+`;
+
+const StyledTabFlexItem = styled.div`
+  text-overflow: ellipsis;
+  white-space: nowrap;
+  overflow: hidden;
+`;
+
+const StyledButtonGroup = styled(EuiButtonGroup)`
+  button[data-test-subj='incompatibleTab'] {
```

Comment:
nit: consider alternatives to referencing `data-test-subj`s in styles

Reply:
why? and what alternative do you see here?

Reply:
`data-test-subj` is a reliable selector, but it's _uncommon_ to see it used as a selector in _production_ code in Kibana.

How uncommon?

The following `grep` finds code (in `$KIBANA_HOME`) that uses `data-test-subj` as a `button` selector:

```sh
➜  kibana git:(attack_discovery_output_token_limits) ✗ grep -r --include="*.test.tsx" 'button\[data-test-subj=' .
./packages/home/sample_data_card/src/footer/remove_footer.test.tsx:      component.find(`button[data-test-subj="removeSampleDataSet${id}"]`).simulate('click');
./packages/home/sample_data_card/src/footer/remove_footer.test.tsx:      component.find(`button[data-test-subj="removeSampleDataSet${id}"]`).simulate('click');
./packages/home/sample_data_card/src/footer/install_footer.test.tsx:      component.find(`button[data-test-subj="addSampleDataSet${id}"]`).simulate('click');
./packages/home/sample_data_card/src/footer/install_footer.test.tsx:      component.find(`button[data-test-subj="addSampleDataSet${id}"]`).simulate('click');
# ... output truncated for brevity
```

when the output above is piped to `wc`, we can count the usages to see if it's common.

There are `329` usages In just _test_ files `*.test.tsx` :

```sh
➜  kibana git:(feat/DQD/185881-add-list-view-and-latest-flyout) ✗ grep -r --include="*.test.tsx" 'button\[data-test-subj=' . | wc -l
     329
```

There are `331` usages when we also include _production_ files `*.tsx` :

```sh
➜  kibana git:(feat/DQD/185881-add-list-view-and-latest-flyout) ✗ grep -r --include="*.tsx" 'button\[data-test-subj=' . | wc -l
     331
```

Where do the additional two production usages come from?

```sh
➜  kibana git:(feat/DQD/185881-add-list-view-and-latest-flyout) ✗ diff <(grep -r --include="*.test.tsx" 'button\[data-test-subj=' .) <(grep -r --include="*.tsx" 'button\[data-test-subj=' .)
316a317,318
> ./x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_properties/index_check_fields/index.tsx:  button[data-test-subj='incompatibleTab'] {
> ./x-pack/packages/security-solution/ecs_data_quality_dashboard/impl/data_quality/data_quality_panel/index_properties/index_check_fields/index.tsx:  button[data-test-subj='ecsCompliantTab'] {
```

Per the `diff` output above, the only two _production_ usages found are the changes proposed in this PR.

It's ok if there's no better alternative, as this may be the first time it's _required_ (as a button selector) in Kibana's production code.

Reply:
I would argue if we are assessing the usage of `data-test-subj` in styles better grep that specifically like

`grep -r --include="*.tsx" --exclude="*.test.ts*" '\[data-test-subj.*\] {' -A1`

![image](https://github.com/user-attachments/assets/a25d2ec6-e03e-4da3-9446-6720f003d102)

As seen in the screenshot there is more usage than introduced in this PR. 

Of course it's not common, but I am not sure commonality is a strong factor, even tho it's a factor. I would say as a general rule as long as it makes sense and doesn't bring more problems than it solves including stylistic ones, why not? Wdyt?

In this specific case I don't see a more elegant hook into button tab within button group. But feel free to share if you see.